### PR TITLE
WIP: Switch to directly using MeshCat for vis and InteractBase for buttons

### PR DESCRIPTION
**Pull request by rdeits**
Todo:

- [x] Fix tests
- [x] Fix notebooks
- [x] Display the current sim time in the control widget
- [x] Pausing and terminating animation
- [x] Make controls widget look nicer
- [x] Check documentation

I haven't had time to get this all polished up yet, but this is a sketch of the idea in #45. This completely splits off the pause/terminate controls from the visualizer (hypothetically we could put the buttons in the visualizer later, but it's kind of a challenge to get right). Also, being able to pause a simulation without having to spawn an entire 3D visualizer is kind of nice. 

Usage is roughly: 

```julia
using Blink

controls = SimulationControls()
open(controls, Window())  # create a new Blink window to hold the controls
callback = CallbackSet(controls)
```

which can be combined with the visualizer callback, rate limiter, etc. 

This drops the entire RigidBodyTreeInspector/LCMCore/DrakeVisualizer tree from the dependencies. I added Blink as a dependency because standalone windows are really nice for this kind of thing. However, we *could* avoid depending on Blink and just render things in the browser. But I think this is nicer. 

This also will make it possible in the future to render simulation controls in Juno or Jupyter.

### CODE REVIEW

**Comment by rdeits** [comment]
This also hooks into the new MeshCat animation system, so you can do: 

```julia
setanimation!(vis, sol::ODESolution)
```

and have the animation rendered nicely (with play/pause and speed controls) in the MeshCat viewer.

**Comment by tkoolen** [comment]
Thanks so much! I was starting to look into how to do this, but was feeling a bit daunted. 

I did run into:

```
MethodError: Cannot `convert` an object of type Observables.Observable{Any} to an object of type Observables.Observable{Bool}
This may have arisen from a call to the constructor Observables.Observable{Bool}(...),
since type constructors fall back to convert methods.

Stacktrace:
 [1] RigidBodySim.Visualization.SimulationStatus(::RigidBodySim.Visualization.SimulationControls{Widgets.Widget{:button}}) at /Users/twan/code/julia/RigidBodySim/v0.6/RigidBodySim/src/visualization.jl:172
 [2] DiffEqBase.CallbackSet(::RigidBodySim.Visualization.SimulationControls{Widgets.Widget{:button}}) at /Users/twan/code/julia/RigidBodySim/v0.6/RigidBodySim/src/visualization.jl:175
```

when running the code in the PR description.

Eventually, I do really want to have the simulation controls and the 3D canvas in one window.

Let me know if there's anything I can help with.

File: deps/build.jl [hunk 1]
```diff
@@ -0,0 +1,2 @@
+using Blink
+Blink.AtomShell.isinstalled() || Blink.AtomShell.install()
```
Comment (tkoolen) [inline deps/build.jl]: Why is this not part of Blink's build script?

File: src/visualization.jl [hunk 2]
```diff
@@ -78,127 +79,166 @@ using DocStringExtensions
...
-            copy!(state, integrator.u)
-            visualize(vis, integrator.t, state)
+            copy!(vis, integrator.u)
             u_modified!(integrator, false)
         end
     end
     DiscreteCallback(condition, action; save_positions = (false, false))
 end
 
+@deprecate CallbackSet(vis, state::MechanismState; max_fps = 60.) CallbackSet(vis, state; max_fps = max_fps)
```
Comment (tkoolen) [inline src/visualization.jl]: This doesn't seem right?

File: src/visualization.jl [hunk 3]
```diff
@@ -78,127 +79,166 @@ using DocStringExtensions
...
+
+struct SimulationControls{W <: Widget}
```
Comment (tkoolen) [inline src/visualization.jl]: Why parameterize this? Why not have `terminate::Widget{:button}`?

File: src/visualization.jl [hunk 4]
```diff
@@ -78,127 +79,166 @@ using DocStringExtensions
...
+    terminate::W
+    pause::W
+end
+
+SimulationControls() = SimulationControls(button("terminate"), button("pause"))
```
Comment (tkoolen) [inline src/visualization.jl]: Is there a way to get the pause button to stay depressed when the simulation is paused?

File: deps/build.jl (hunk 1 above)
Comment (rdeits) [reply deps/build.jl]: Yeah, I wish it were.

**Comment by rdeits** [comment]
I think you'll need `MeshCatMechanisms` master to get that MethodError to go away. 

I'd also like to have an option to put buttons in the visualizer, but I think the way to do that is by composition rather than modifying MeshCat itself. What I mean is, now that everything is just an HTML element, we can combine them like any other HTML elements. So you might be able to do something like: 

```julia
window = Window()
body!(window, vbox(controls, visualizer))
```

to create a single window holding everything. That won't quite work now, but I don't think it's too hard.

File: src/visualization.jl (hunk 2 above)
Comment (rdeits) [reply src/visualization.jl]: whoops, thanks

**Comment by tkoolen** [comment]
I'm working on the notebook.

**Comment by rdeits** [comment]
Actually MeshCat already knows how to do that (but the discoverability is bad). Just click the "show controls" drop-down in the top right. There's an "Animations" section with play/pause/reset and speed controls.

**Comment by tkoolen** [comment]
Ideally, the same buttons should be usable to control animation and simulation, but it's OK as it is for now if that's hard.

**Comment by tkoolen** [comment]
Dumb Travis OSX failure due to #62. I'm going to merge now and fix that later. Thanks for your help!

**Pull request merged**